File: 9/9.clj
(defn square [x] (Math/pow x 2))
(defn is-pythagorean?
  [a b c]
  (= (+ (square a) (square b)) (square c)))

(println
  (loop [a 1 b 1]
    (let [c (- 1000 a b)]
    (println a b c)
    (cond
      (is-pythagorean? a b c)
      (* a b c)
      (not (<= a b c))
      (recur (inc a) (inc a))
      :else
      (recur a (inc b))))))

File: 9/9.rs
fn is_pythagorean_triplet(a: u32, b: u32, c: u32) -> bool {
    return a.pow(2) + b.pow(2) == c.pow(2);
}

fn main() {
    let (mut a, mut b) = (1, 1);
    while a + b < 1000 {
        let c = 1000 - a - b;
        if is_pythagorean_triplet(a, b, c) {
            println!("{}", a * b * c);
            break;
        }
        if b < c {
            b += 1;
        } else {
            a += 1;
            b = 1;
        }
    }
}
